.github/workflows/on_pull_request_synchronized.yml

```diff
@@ -4,6 +4,18 @@ on:
     types:
       - synchronize
 jobs:
+  trigger-premerge:
+    runs-on: ubuntu-latest
+    if: contains(github.event.pull_request.labels.*.name, 'go')
+    steps:
+      - uses: buildkite/trigger-pipeline-action@v2.2.0
+        with:
+          buildkite_api_access_token: ${{ secrets.TRIGGER_BK_BUILD_TOKEN }}
+          pipeline: "ray-project/premerge"
+          branch: ${{ github.event.pull_request.head.ref }}
+          commit: ${{ github.event.pull_request.head.sha }}
+          pull_request_base_branch: ${{ github.event.pull_request.base.ref }}
```

Comment:
this should be done on pipeline config. buildkite pipeline can be configured to only be triggered on certain PR's with certain labels or conditions.

we want to minimize the usage of github actions. we do not want to trigger buildkite using github actions. and also don't want to give people access a buildkite api access token.